libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -337,6 +337,68 @@ MAV_RESULT AP_Mount::handle_command_do_gimbal_manager_pitchyaw(const mavlink_com
     return MAV_RESULT_FAILED;
 }
 
+void AP_Mount::handle_gimbal_manager_set_attitude(const mavlink_message_t &msg){
+    mavlink_gimbal_manager_set_attitude_t packet;
+    mavlink_msg_gimbal_manager_set_attitude_decode(&msg,&packet);
+
+    AP_Mount_Backend *backend;
+
+    // check gimbal device id.  0 is primary, 1 is 1st gimbal, 2 is
+    // 2nd gimbal, etc
+    const uint8_t instance = packet.gimbal_device_id;
+    if (instance == 0) {
+        backend = get_primary();
+    } else {
+        backend = get_instance(instance - 1);
+    }
+
+    if (backend == nullptr) {
+        return;
+    }
+
+    // check flags for change to RETRACT
+    uint32_t flags = (uint32_t)packet.flags;
+    if ((flags & GIMBAL_MANAGER_FLAGS_RETRACT) > 0) {
+        backend->set_mode(MAV_MOUNT_MODE_RETRACT);
+        return;
+    }
+
+    // check flags for change to NEUTRAL
+    if ((flags & GIMBAL_MANAGER_FLAGS_NEUTRAL) > 0) {
+        backend->set_mode(MAV_MOUNT_MODE_NEUTRAL);
+        return;
+    }
+
+    if (!isnan(packet.q[0]) && !isnan(packet.q[1]) && !isnan(packet.q[2]) && !isnan(packet.q[3])) {
+        // convert quaternion to euler angles
+        Quaternion att_quat;
+        att_quat.q1 = packet.q[0];
+        att_quat.q2 = packet.q[1];
+        att_quat.q3 = packet.q[2];
+        att_quat.q4 = packet.q[3];
+
+        // quaternion to eular conversion
+        float roll_rad, pitch_rad, yaw_rad;
+        att_quat.to_euler(roll_rad, pitch_rad, yaw_rad);
+
+        // radian to deg conversion
+        float roll_deg, pitch_deg, yaw_deg;
+        roll_deg = RadiansToCentiDegrees(roll_rad)*100.0f;
+        pitch_deg = RadiansToCentiDegrees(pitch_rad)*100.0f;
+        yaw_deg = RadiansToCentiDegrees(yaw_rad)*100.0f;
```

Comment:
```suggestion
        const float roll_deg = degrees(roll_rad);
        const float pitch_deg = degrees(pitch_rad);
        const float yaw_deg = degrees(yaw_rad);
```

similarly on the angular rates